systemtests/src/test/java/io/enmasse/systemtest/common/plans/PlansTest.java

```diff
@@ -307,6 +309,184 @@ public void testGlobalSizeLimitations() throws Exception {
+                new AddressSpaceResource("broker", 0.0, 5.0),
+                new AddressSpaceResource("router", 1.0, 5.0),
+                new AddressSpaceResource("aggregate", 0.0, 5.0));
+        List<AddressPlan> addressPlans = Arrays.asList(queuePlanDistributed, queuePlanSharded);
+        AddressSpacePlan scaleSpacePlan = new AddressSpacePlan("scale-plan", "scale-plan",
+                "standard-space", AddressSpaceType.STANDARD, resources, addressPlans);
+        plansProvider.createAddressSpacePlanConfig(scaleSpacePlan);
+
+        //create address space plan with new plan
+        AddressSpace messagePersistAddressSpace = new AddressSpace("persist-messages-space-standard", AddressSpaceType.STANDARD,
+                scaleSpacePlan.getName(), AuthService.STANDARD);
+        createAddressSpace(messagePersistAddressSpace);
+
+        //deploy destinations
+        Destination queue = Destination.queue("distributed-queue", queuePlanDistributed.getName());
+        setAddresses(messagePersistAddressSpace, queue);
+
+        //pod should have 2 replicas
+        TestUtils.waitForNBrokerReplicas(kubernetes, messagePersistAddressSpace.getNamespace(), 2, queue, new TimeoutBudget(2, TimeUnit.MINUTES));
+
+        //send 100000 messages to queue
+        String username = "test_change_plan_user";
+        String password = "test_change_plan_pswd";
+        createUser(messagePersistAddressSpace, username, password);
+
+        AmqpClient queueClient = amqpClientFactory.createQueueClient(messagePersistAddressSpace);
+        queueClient.getConnectOptions().setUsername(username);
+        queueClient.getConnectOptions().setPassword(password);
+
+        List<String> msgs = TestUtils.generateMessages(100_000);
+        Future<Integer> sendResult1 = queueClient.sendMessages(queue.getAddress(), msgs);
+        assertThat("Incorrect count of messages sent", sendResult1.get(1, TimeUnit.MINUTES), is(msgs.size()));
+
+        //replace original plan in address by another
+        replaceAddressPlan(messagePersistAddressSpace, queue, queuePlanSharded);
+
+        assertEquals(getAddressesObjects(
+                messagePersistAddressSpace,
+                Optional.of(queue.getAddress())).get(10, TimeUnit.SECONDS).get(0).getPlan(),
+                queuePlanSharded.getName(),
+                "New plan wasn't set correctly");
+
+        //wait until address will be scaled down to 1 pod
+        TestUtils.waitForNBrokerReplicas(
+                kubernetes,
+                messagePersistAddressSpace.getNamespace(), 1, queue, new TimeoutBudget(2, TimeUnit.MINUTES));
+        //test failed in command above ^, functionality of test code below wasn't verified :) !TODO
+
+        //receive messages
+        Future<List<Message>> recvResult = queueClient.recvMessages(queue.getAddress(), msgs.size());
+        assertThat("Incorrect count of messages received", recvResult.get(1, TimeUnit.MINUTES).size(), is(msgs.size()));
+    }
+
```

Comment:
Please close the clients :)

Reply:
Aren't clients closed automatically in TestBase? https://github.com/EnMasseProject/enmasse/blob/master/systemtests/src/test/java/io/enmasse/systemtest/bases/TestBase.java#L104
This shouldn't be different case and "default" tearDown should cover it. Correct me if I'm wrong please.